packages/collector/test/tracing/database/ioredis/connect-via/cluster.js

```diff
@@ -0,0 +1,51 @@
+/*
+ * (c) Copyright IBM Corp. 2024
+ */
+
+'use strict';
+
+// NOTE: We run the tests locally and on CI against azure redis cluster.
+// NOTE: We cannot run redis cluster on Tekton https://github.com/bitnami/charts/issues/28894
+// NOTE: We cannot use a docker based redis cluster at the moment!
+//       See https://github.com/redis/node-redis/issues/2815.
+//       These commands are useful as soon as we switch to a docker based cluster.
+//       node bin/start-test-containers.js --redis-node-0 --redis-node-1 --redis-node-2
+//       docker exec -it 2aaaac7b9112 redis-cli -p 6379 cluster info
+module.exports = async function connect(ioredis, log) {
+  if (!process.env.AZURE_REDIS_CLUSTER || !process.env.AZURE_REDIS_CLUSTER_PWD) {
```

Comment:
While running the tests locally, I initially encountered the following issue:

> [ioredis] Unhandled error event: ClusterAllFailedError: Failed to refresh slots cache.

The issue seems to have resolved itself after some time. It might be related to the cluster since we use the same cluster for both Redis and ioredis testing. Running tests in parallel could potentially cause conflicts. We should consider alternative solutions, such as using namespaces, to prevent this issue in the future.

Reply:
https://jsw.ibm.com/browse/INSTA-14042